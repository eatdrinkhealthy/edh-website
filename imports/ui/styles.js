import { css } from "glamor";

//
// Component Style Pattern (use with glamor)
//
// const componentRules = {
//    rule1: "value",
//      ...
//    ruleN: "value",
// };
// export const componentStyle = {
//    rules: componentRules,
//    className: `${css(componentRules)}`,
// };
//

const appLinksHeaderRules = {
  marginTop: "90px",
  marginLeft: "40px",
  fontSize: "16px",
  color: "darkbrown",
};
export const appLinksHeaderStyle = {
  rules: appLinksHeaderRules,
  className: `${css(appLinksHeaderRules)}`,
};

const appLinksButtonsRules = {
  marginTop: "14px",
  marginBottom: "20px",
  marginLeft: "40px",
};
export const appLinksButtonsStyle = {
  rules: appLinksButtonsRules,
  className: `${css(appLinksButtonsRules)}`,
};

const linkButtonRules = {
  backgroundColor: "darkslategray",
  border: "none",
  borderRadius: "4px",
  color: "white",
  padding: "8px 32px 8px 32px",
  marginRight: "8px",
  textAlign: "center",
  textDecoration: "none",
  fontSize: "12px",
};
export const linkButtonStyle = {
  rules: linkButtonRules,
  className: `${css(linkButtonRules)}`,
};
