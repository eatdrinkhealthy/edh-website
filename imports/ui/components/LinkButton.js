import React from "react";
import { linkButton } from "../styles";

const LinkButton = () => (
  <div>
    <a href="#" className={linkButton}>LinkButton</a>
  </div>
);

export default LinkButton;
