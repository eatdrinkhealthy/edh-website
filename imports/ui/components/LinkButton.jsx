import React from "react";
import PropTypes from "prop-types";
import { linkButtonStyle } from "../styles";

const LinkButton = props => (
  <a href={props.url} className={linkButtonStyle.className}>
    {props.text}
  </a>
);

LinkButton.propTypes = {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default LinkButton;
