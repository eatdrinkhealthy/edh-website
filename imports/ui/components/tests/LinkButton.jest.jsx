/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
import React from "react";
import renderer from "react-test-renderer";
import LinkButton from "../LinkButton";
import {
  formattedJSON,
  getAllComponentStyle,
} from "../../../utils/tests/testUtils";
import * as styles from "../../../ui/styles";

describe("<LinkButton />", function () {
  it("matches render snapshot", function () {
    const tree = renderer.create(<LinkButton text="some button text" url="some link" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches style / css snapshot", function () {
    const allStyle = getAllComponentStyle(<LinkButton text="button text" url="http..." />, styles);
    expect(formattedJSON(allStyle)).toMatchSnapshot();
  });
});
