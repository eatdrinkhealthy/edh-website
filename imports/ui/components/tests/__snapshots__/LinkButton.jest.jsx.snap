// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> matches render snapshot 1`] = `
<a
  className="css-1trporh"
  href="some link"
>
  some button text
</a>
`;

exports[`<LinkButton /> matches style / css snapshot 1`] = `
"[
  {
    \\"rules\\": {
      \\"backgroundColor\\": \\"darkslategray\\",
      \\"border\\": \\"none\\",
      \\"borderRadius\\": \\"4px\\",
      \\"color\\": \\"white\\",
      \\"padding\\": \\"8px 32px 8px 32px\\",
      \\"marginRight\\": \\"8px\\",
      \\"textAlign\\": \\"center\\",
      \\"textDecoration\\": \\"none\\",
      \\"fontSize\\": \\"12px\\"
    },
    \\"className\\": \\"css-1trporh\\"
  }
]"
`;
